### PR TITLE
7994 ga events view dependents

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
[Original Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7994)

This ticket is to add the analytics events for the View Dependents page. The events needed are

1. The event for when the page loads and is successful
2. The event for when the page loads an fails
3. The event for when someone clicks the primary button on the View Dependents page

## Testing done
Tested that all events fire using dataslayer in Firefox

## Acceptance criteria
- [x] Events fire with the appropriate names

### CODE REVIEW

File: src/applications/personalization/view-dependents/actions/index.js [hunk 1]
```diff
@@ -8,11 +9,16 @@ export function fetchAllDependents() {
     const response = await getData('/dependents_applications/show');
 
     if (response.errors) {
+      recordEvent({
+        event: `disability-view-dependents-load-failed`,
+        'error-key': `${response.errors[0].status}_internal_error`,
```
Comment (micahchiang) [inline src/applications/personalization/view-dependents/actions/index.js:14]: Are the underscores necessary here? Looks like this ticket suggests an error-key without them - https://github.com/department-of-veterans-affairs/va.gov-team/issues/6830

File: src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+      <div className="vads-l-row">
+        <div className="vads-l-col--12">
+          <h1>Your VA Dependents</h1>
+          <p className="vads-u-font-size--md vads-u-font-family--serif">
+            Below is a list of dependents we have on file for you. You can file
+            a claim for additional disability compensation whenever you add a
+            new dependent.
+          </p>
+          <a
+            href="https://www.ebenefits.va.gov/ebenefits/about/feature?feature=dependent-compensation"
```
Comment (micahchiang) [inline src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx:22]: I know it's unrelated to this ticket, but if view-dependents is launching beside the 686, should this URL just be updated?

**Review by micahchiang — APPROVED** [review]
LGTM. Just left a couple questions.

File: src/applications/personalization/view-dependents/actions/index.js (hunk 1 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/actions/index.js:14]: Jon, the analytics guy, said he would rather have the errors with underscores in Slack https://dsva.slack.com/archives/GV9H4FZGB/p1587667608005200

File: src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx (hunk 2 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx:22]: we have a separate ticket to update the URLs for the launch

File: src/applications/personalization/view-dependents/actions/index.js [hunk 3]
```diff
@@ -8,11 +9,16 @@ export function fetchAllDependents() {
...
+      });
       dispatch({
         type: FETCH_ALL_DEPENDENTS_FAILED,
         response,
       });
     } else {
+      recordEvent({ event: `disability-view-dependents-load-success` });
```
Comment (ncksllvn) [inline src/applications/personalization/view-dependents/actions/index.js:21]: Nitpick but you don't need the template string here

File: src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx [hunk 4]
```diff
@@ -1,22 +1,34 @@
...
-        Add or remove a dependent
-      </a>
-    </div>
-  </div>
-);
+class ViewDependentsHeader extends Component {
+  handleClick = () => {
+    recordEvent({
+      event: 'cta-default-button-click',
+    });
```
Comment (ncksllvn) [inline src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx:8]: Not necessary at all right now but if we find ourselves making a bunch of CTA buttons to eBenefits we may want to make a super simple component to abstract the GA event and maybe enforce the URL structure (`?feature=...`)

**Review by ncksllvn — APPROVED** [review]
LGTM!

File: src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx (hunk 4 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/components/ViewDependentsHeader/ViewDependentsHeader.jsx:8]: I will bring this up with the Analytics team

**Pull request merged**